From 6a7cd7001fd831ac1a93d1b769c70d16359b3c14 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Mon, 23 Jan 2023 20:48:01 +0000
Subject: [PATCH 1074/1077] MANTIS 0063778 extension: Firealarm handling in DTM

codes.idl
V1.5.3
- added status codes BAGGAGE_FIRE_ALARM

platform.idl
V5.10
add parameter hard for enableDev/disableDev
add enum entry ConveyorState.CON_FIRE_ALARM


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1760 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/idl/Cuss/codes.idl      | 8 +++++++-
 SelfServiceCommon/Massai/idl/Massai/platform.idl | 7 ++++---
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Cuss/codes.idl b/SelfServiceCommon/Massai/idl/Cuss/codes.idl
index e63076d9..720976f4 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/codes.idl
+++ b/SelfServiceCommon/Massai/idl/Cuss/codes.idl
@@ -6,7 +6,7 @@
 //
 //  Date:      31.03.2015
 //
-//  Version:   1.5.0
+//  Version:   1.5.3
 //
 //  Author:    IATA Passenger Experience Management Group: CUWG CUSS-TSG
 //
@@ -108,6 +108,11 @@ Changed:   Added baggage handling status codes (Nicolas Bruns, 29.06.2004)
 @li Mantis 0061647: Conveyor-Component: new mode for conveyor: maintenance, forceBag, off
 - added status codes BAGGAGE_MAINTENANCE_MODE, BAGGAGE_FORCEBAG_MODE, BAGGAGE_OFF_MODE
 
+@date 08.10.2021
+@li V1.5.3
+@li Mantis 0063778: Firealarm handling in DTM
+- added status codes BAGGAGE_FIRE_ALARM
+
 */
 
 #ifndef CODES_IDL
@@ -408,6 +413,7 @@ module statuscodes
     const long BAGGAGE_MAINTENANCE_MODE      = 177; /**< <b>Scope:</b> public; materna den extension */
     const long BAGGAGE_FORCEBAG_MODE         = 178; /**< <b>Scope:</b> public; materna den extension */
     const long BAGGAGE_OFF_MODE              = 179; /**< <b>Scope:</b> public; materna den extension */
+    const long BAGGAGE_FIRE_ALARM            = 180; /**< <b>Scope:</b> public; materna dtm extension */
 
 
     const long FORMAT_ERROR                = 201; /**< <b>Scope:</b> private + platform (platform only for output classes) */
diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 3142dca6..4dcd1862 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -27,7 +27,7 @@
 
 module PlatformManagement
 {
-    const string interface_version = "5.9";
+    const string interface_version = "5.10";
 
     /** Platform interface for the device components. Via this interface the device
         components communicate with the platform manager internally. */
@@ -191,8 +191,8 @@ module PlatformManagement
         */
         code resetMediaRecord(in mRealComponents mediaOffRoller);
 
-        code enableDevice();
-        code disableDevice();
+        code enableDevice(in boolean hard);
+        code disableDevice(in boolean hard);
         code switchEnableDevice(out boolean on);
 
         code checkBindingConflict();
@@ -379,6 +379,7 @@ module PlatformManagement
       CON_ERROR,
       CON_MAINTENANCE,
       CON_FORCEBAG,
+      CON_FIRE_ALARM,
       CON_OFF
     };
 
-- 
2.41.0.windows.1

